### PR TITLE
Linux windows hybrid deploy

### PR DESCRIPTION
**Pull request by Lyndon-Li**
Fix issue #8416, #8417, deploy Velero server and node-agent in linux/Windows hybrid env:

- Velero server is restricted to run on linux nodes only
- node-agent is divided into two instances. node-agent instance runs in linux nodes only and node-agent-windows runs in Windows only
- node-agent daemonset is deployed when `--use-node-agent `is specified; node-agent-windows daemonset is deployed when `--use-node-agent-windows` is specified
- In a pure linux nodes cluster, `desiredNumberScheduled` for node-agent-windows is 0
- In a pure Windows nodes cluster, `desiredNumberScheduled` for node-agent is 0
- In a hybrid cluster, `desiredNumberScheduled` for node-agent-windows is the number of Windows nodes; `desiredNumberScheduled` for node-agent is the number of linux nodes

### CODE REVIEW

File: pkg/cmd/cli/nodeagent/server.go [hunk 1]
```diff
@@ -200,13 +200,31 @@ func newNodeAgentServer(logger logrus.FieldLogger, factory client.Factory, confi
 			},
 		},
 	}
-	mgr, err := ctrl.NewManager(clientConfig, ctrl.Options{
-		Scheme: scheme,
-		Cache:  cacheOption,
-	})
+
+	var mgr manager.Manager
```
Comment (Lyndon-Li) [inline pkg/cmd/cli/nodeagent/server.go:204]: This is to resolve the problem that in some Windows env, the pod network is not immediately ready. Without this, the pod may restart for several times

**Comment by Lyndon-Li** [comment]
Technically, we don't need to distinguish different OS types for install, node-agent and node-agent-windows ds instance could be installed in both OS all the time.
I don't see any problem about this, but let me know if you have any concern about the same. @sseago @shubham-pampattiwar @kaovilai @reasonerjt @blackpiglet

**Comment by kaovilai** [comment]
I don't see a problem with 0 scheduled daemonsets. It's not taking up any major resources.

File: pkg/install/resources.go [hunk 2]
```diff
@@ -418,6 +418,13 @@ func AllResources(o *VeleroOptions) *unstructured.UnstructuredList {
 		if err := appendUnstructured(resources, ds); err != nil {
 			fmt.Printf("error appending DaemonSet %s: %s\n", ds.GetName(), err.Error())
 		}
+
+		dsOpts = append(dsOpts, WithForWinows())
```
Comment (ywk253100) [inline pkg/install/resources.go:422]: `WithForWinows()` typo

**Comment by reasonerjt** [comment]
Thanks @Lyndon-Li 
I have 2 overall comments:
1) I think we can add some label to make sure velero-server is always running on linux worker-node.
2) We should add a flag like " --windows-support" and only when this flag is true we deploy the daemonset for Windows so by default everything stays the same.

**Comment by Lyndon-Li** [comment]
> Thanks @Lyndon-Li I have 2 overall comments:
> 
> 1. I think we can add the labels/annotations to make sure velero-server is always running on Linux worker-node.
> 2. We should add a flag like " --windows-support" and only when this flag is true we deploy the daemonset for Windows so by default everything stays the same.

As the discussion:
For 1, we will force Velero to run in linux nodes only. We will add a limitation to release notes --- Velero requires at least one linux worker node in order to run with Windows nodes.
For 2, we will add a flag --use-node-agent-windows to install node-agent-windows daemonset.

File: pkg/install/resources.go (hunk 2 above)
Comment (Lyndon-Li) [reply pkg/install/resources.go]: Done

**Review by ywk253100 — APPROVED** [review]

**Review by reasonerjt — APPROVED** [review]

**Pull request merged**